File: app/src/main/java/com/karthik/myweather/network/WeatherService.java
package com.karthik.myweather.network;

import com.karthik.myweather.network.model.ConsolidatedWeather;
import com.karthik.myweather.network.model.Location;
import com.karthik.myweather.network.model.LocationWeather;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface WeatherService {

    @GET("location/search/")
    Single<List<Location>> getQueryResults(@Query("query") String query);

    @GET("location/{location}/")
    Single<ConsolidatedWeather> getConsolidatedWeather(@Path("location") long locationId);

    @GET("location/search/")
    Single<List<Location>> getLocationResults(@Query("lattlong") String latLong);

    @GET("location/{location}/{date}")
    Single<List<LocationWeather>> getConsolidatedWeather(
            @Path("location") long locationId, @Path("date") String date);
}

File: app/src/main/java/com/karthik/myweather/network/WeatherService.kt
package com.karthik.myweather.network

import com.karthik.myweather.network.model.ConsolidatedWeather
import com.karthik.myweather.network.model.Location
import com.karthik.myweather.network.model.LocationWeather
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface WeatherService {
    @GET("location/search/")
    fun getQueryResults(@Query("query") query: String?): Single<List<Location?>?>?

    @GET("location/{location}/")
    fun getConsolidatedWeather(@Path("location") locationId: Long): Single<ConsolidatedWeather?>?

    @GET("location/search/")
    fun getLocationResults(@Query("lattlong") latLong: String?): Single<List<Location?>?>?

    @GET("location/{location}/{date}")
    fun getConsolidatedWeather(
            @Path("location") locationId: Long, @Path("date") date: String?): Single<List<LocationWeather?>?>?
}